src/pipx/venv.py

```diff
@@ -292,11 +314,11 @@ def run_app(self, app: str, app_args: List[str]) -> int:
         except KeyboardInterrupt:
             return 130  # shell code for Ctrl-C
 
-    def _upgrade_package_no_metadata(
-        self, package_or_url: str, pip_args: List[str]
-    ) -> None:
-        with animate(f"upgrading package {package_or_url!r}", self.do_animation):
-            self._run_pip(["install"] + pip_args + ["--upgrade", package_or_url])
+    def _upgrade_package_no_metadata(self, package: str, pip_args: List[str]) -> None:
+        with animate(
+            "upgrading " + full_package_description(package, package), self.do_animation
```

Comment:
nit: can f-strings be used instead of adding strings together (applies elsewhere in this PR as well)?

Reply:
Done.